[option parser] Add conversion negation for thumbnails and subtitles

kieraneglin opened the pull request:
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

<!--

Explanation of your *pull request* in arbitrary form goes here. Please **make sure the description explains the purpose and effect** of your *pull request* and is worded well enough to be understood. Provide as much **context and examples** as possible

-->

(Discussed in Discord)

When using multiple configuration files or generating `yt-dlp` config programmatically, it's beneficial for each command to have an equal and opposite command to negate it.

This PR introduces a new `none` option for both `--convert-thumbnails` and `--convert-subs`. These options entirely prevent their associated post-processing step from being run


<details open><summary>Template</summary> <!-- OPEN is intentional -->

<!--

# PLEASE FOLLOW THE GUIDE BELOW

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes `[ ]` relevant to your *pull request* (like [x])
- Use *Preview* tab to see how your *pull request* will actually look like

-->

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [x] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details>

Grub4K commented:
The handling of the `none` value should be strictly done in `options.py`/`__init__.py`, resulting in no postprocessor being added at all

kieraneglin commented:
@Grub4K If that's the preference, that would make things much more simple! I'll get on that change shortly

kieraneglin commented:
@Grub4K updated! To fully extract the logic out of the post-processors I had to create a regex for `thumbnail format` inline in `__init__.py`. Let me know what you think!

kieraneglin commented on yt_dlp/__init__.py:
```diff
@@ -621,13 +621,13 @@ def get_postprocessors(opts):
             'when': 'after_filter',
         }
-    if opts.convertsubtitles:
+    if opts.convertsubtitles and opts.convertsubtitles != 'none':
```
I could also approach this with something like:

```python
if opts.convertsubtitles == 'none':
  opts.convertsubtitles = None
```

earlier in the file. This would also remove the need for my regex changes, but arguably it makes the code harder to follow. Let me know your preference!

Grub4K replied:
i think doing
```py
if opts.convertsubtitles == 'none':
  opts.convertsubtitles = None
```
for both should be the easiest

Grub4K reviewed (commented):
Dont forget to update the values in `options.py`. The readme is automatically generated from the output of `--help`

kieraneglin commented:
@Grub4K I didn't realize that! Updated. I also changed `convertsubtitles` to use an f-string like `convertthumbnails` since that cleaned things up

![Screenshot 2024-09-23 at 12 04 28 PM](https://github.com/user-attachments/assets/c5a87ae3-c1ae-45a5-b540-d0ed9b4b9150)

bashonly commented on yt_dlp/options.py:
```diff
@@ -1725,14 +1725,15 @@ def _alias_callback(option, opt_str, value, parser, opts, nargs):
         help=(
             'Convert the thumbnails to another format '
-            f'(currently supported: {", ".join(sorted(FFmpegThumbnailsConvertorPP.SUPPORTED_EXTS))}). '
+            f'(currently supported: {", ".join(sorted(FFmpegThumbnailsConvertorPP.SUPPORTED_EXTS))}, none). '
             'You can specify multiple rules using similar syntax as --remux-video'))
```
tacking `none` on to the end of the extension list is misleading here IMO, since the following line then implies you can use it like `--convert-thumb "jpg>png/webp>none"`
```suggestion
            f'(currently supported: {", ".join(sorted(FFmpegThumbnailsConvertorPP.SUPPORTED_EXTS))}). '
            'You can specify multiple rules using similar syntax as "--remux-video". '
            'A previous pass of this option can be negated with "--convert-thumbnails none"'))
```

kieraneglin replied:
Fair enough! Even though subtitles don't have the same caveat, would you like me to use the same language there for consistency?

bashonly replied:
If you want to. I'm not exactly thrilled with "a previous pass of this option" as the wording, though

kieraneglin replied:
It's hard to convey succinctly! Thoughts on `Use "--convert-thumbnails none" to disable conversion (default)`? This would rely on the user knowing that options will be overwritten if they're repeated

bashonly replied:
```suggestion
            f'(currently supported: {", ".join(sorted(FFmpegThumbnailsConvertorPP.SUPPORTED_EXTS))}). '
            'You can specify multiple rules using similar syntax as "--remux-video". '
            'Use "--convert-thumbnails none" to disable conversion (default)'))
```

bashonly commented on yt_dlp/options.py:
```diff
@@ -1725,14 +1725,15 @@ def _alias_callback(option, opt_str, value, parser, opts, nargs):
-            'Convert the subtitles to another format (currently supported: {}) '
-            '(Alias: --convert-subtitles)'.format(', '.join(sorted(FFmpegSubtitlesConvertorPP.SUPPORTED_EXTS)))))
+            'Convert the subtitles to another format '
+            f'(currently supported: {", ".join(sorted(FFmpegSubtitlesConvertorPP.SUPPORTED_EXTS))}, none) '
+            '(Alias: --convert-subtitles)'))
```
```suggestion
            f'(currently supported: {", ".join(sorted(FFmpegSubtitlesConvertorPP.SUPPORTED_EXTS))}). '
            'Use "--convert-subs none" to disable conversion (default) (Alias: --convert-subtitles)'))
```
